lawnchair/src/app/lawnchair/search/data/SearchRequest.kt

```diff
@@ -0,0 +1,206 @@
+                                val jsonObject = JSONObject()
+                                jsonObject.put(CONTACT_ACCOUNT_ID, key)
+                                jsonObject.put(CONTACT_ACCOUNT_TITLE, data5)
+                                jsonObject.put(CONTACT_ACCOUNT_NAME, accountName)
+                                jsonObject.put(CONTACT_ACCOUNT_TYPE, accountType)
+                                jsonObject.put(CONTACT_ACCOUNT_MIME, mimeType)
+                                jsonArray.put(jsonObject)
+                                existingContact?.packages = jsonArray.toString()
+                            }
+                        }
+                    } catch (e: Exception) {
+                        Log.e("ContactSearch", "Something went wrong ", e)
+                    }
+                }
+            }
+            queryCursor?.close()
+            continuation.resume(ArrayList(contactMap.values))
+        }
+    }
+}
+
+suspend fun findByFileName(context: Context, query: String, max: Int): List<FileInfo> {
+    if (query.isEmpty() || query.isBlank()) return emptyList()
+    return withContext(Dispatchers.IO) {
+        suspendCoroutine { continuation ->
+            val fileList = HashMap<String, FileInfo>()
+
+            val projection = arrayOf(
+                MediaStore.Files.FileColumns._ID,
+                MediaStore.Files.FileColumns.DATE_ADDED,
+                MediaStore.Files.FileColumns.DATA,
+                MediaStore.Files.FileColumns.MEDIA_TYPE,
+                MediaStore.Files.FileColumns.DISPLAY_NAME,
+                MediaStore.Files.FileColumns.MIME_TYPE,
+            )
+
+            val selection = MediaStore.Files.FileColumns.DISPLAY_NAME + " LIKE ?"
+            val selectionArgs = arrayOf("%$query%")
+
+            val sortOrder = MediaStore.Files.FileColumns.DATE_ADDED + " DESC"
+
+            val queryCursor = context.contentResolver.query(
+                MediaStore.Files.getContentUri("external"),
+                projection,
+                selection,
+                selectionArgs,
+                sortOrder,
+            )
+
+            queryCursor?.use {
+                while (it.moveToNext()) {
+                    try {
+                        val fileIdIndex = it.getColumnIndex(MediaStore.Files.FileColumns._ID)
+                        val displayNameIndex = it.getColumnIndex(MediaStore.Files.FileColumns.DISPLAY_NAME)
+                        val mimeTypeIndex = it.getColumnIndex(MediaStore.Files.FileColumns.MIME_TYPE)
+                        val mediaTypeIndex = it.getColumnIndex(MediaStore.Files.FileColumns.MEDIA_TYPE)
+                        val filePathIndex = it.getColumnIndex(MediaStore.Files.FileColumns.DATA)
+
+                        val fileId = it.getLong(fileIdIndex)
+                        val displayName = it.getString(displayNameIndex)
+                        val mimeType = getMimeType(displayName) ?: it.getString(mimeTypeIndex)
```

Comment:
You are using `displayName` here, it should be `extension`. And we have no need to query mime from MediaStore, just use `String.extension2MimeType()`. For extension, we can use:

```kt
internal val Path.extension: String?
  get() {
    val dotIndex = name.lastIndexOf(".")
    return if (dotIndex == -1) null else name.substring(dotIndex + 1)
  }

internal val Path.nameWithoutExtension: String get() = name.substringBeforeLast(".")
```

Reply:
Oh right, but displayName have extension too

Reply:
Explicit extension as a field, and generate mime from it.

Reply:
Right are you able to do it? I have some stuff to do if not I'll do it later after work. Thanks

Reply:
can you implement this? and push to this branch Thanks

Reply:
Yeah, but I'm busy now, will take a look tonight.

Reply:
I'm starting this.